helix-term/src/ui/editor.rs

```diff
@@ -465,12 +474,23 @@ impl EditorView {
 
             let selected = cursors.contains(&line);
 
-            if let Some(_breakpoint) = breakpoints.and_then(|breakpoints| {
-                breakpoints
-                    .iter()
-                    .find(|breakpoint| breakpoint.line == line)
-            }) {
-                surface.set_stringn(viewport.x, viewport.y + i as u16, "▲", 1, warning);
+            if let Some(bps) = breakpoints.as_ref() {
+                if let Some(_) = bps.iter().find(|breakpoint| breakpoint.line == line) {
+                    surface.set_stringn(viewport.x, viewport.y + i as u16, "▲", 1, warning);
+                }
+            }
+
+            if let Some(sp) = stack_pointer.as_ref() {
+                if let Some(src) = sp.source.as_ref() {
+                    if doc
+                        .path()
+                        .and_then(|path| Some(src.path == Some(path.clone())))
+                        .unwrap_or(false)
+                        && sp.line == line
+                    {
+                        surface.set_stringn(viewport.x, viewport.y + i as u16, "⇒", 1, warning);
```

Comment:
We must re-render everything after a new event is received. Should be okay after we dispatch events in editor main loop